src/coreComponents/physicsSolvers/fluidFlow/GeochemicalModel.cpp

```diff
@@ -0,0 +1,819 @@
+
+  std::map<string, string_array > fieldNames;
+  fieldNames["elems"].emplace_back( viewKeyStruct::deltaPressureString );
+  fieldNames["elems"].emplace_back( viewKeyStruct::deltaTemperatureString );
+  fieldNames["elems"].emplace_back( viewKeyStruct::deltaConcentrationString );
+
+  CommunicationTools::SynchronizeFields( fieldNames, &mesh, domain.getNeighbors() );
+
+  this->forTargetSubRegions( mesh, [&] ( localIndex const,
+                                         ElementSubRegionBase & subRegion )
+  {
+    UpdateState( &subRegion );
+  } );
+
+}
+
+void GeochemicalModel::SolveSystem( DofManager const & dofManager,
+                                     ParallelMatrix & matrix,
+                                     ParallelVector & rhs,
+                                     ParallelVector & solution )
+{
+  GEOSX_MARK_FUNCTION;
+
+  rhs.scale( -1.0 );
+  solution.zero();
+
+  SolverBase::SolveSystem( dofManager, matrix, rhs, solution );
+
+  if( getLogLevel() >= 2 )
+  {
+    GEOSX_LOG_RANK("After GeochemicalModel::SolveSystem");
+    GEOSX_LOG_RANK("\nsolution\n");
+    std::cout<<solution<<std::endl;
+  }
+
+}
+
+void GeochemicalModel::ResetStateToBeginningOfStep( DomainPartition & GEOSX_UNUSED_PARAM(domain) )
+{
+}
+
+void GeochemicalModel::ResetViews( MeshLevel & mesh )
+{
+  FlowSolverBase::ResetViews(mesh);
+
+  ElementRegionManager * const elemManager = mesh.getElemManager();
+
+  m_pressure =
+    elemManager->ConstructViewAccessor<array1d<real64>, arrayView1d<real64>>( viewKeyStruct::pressureString );
+  m_deltaPressure =
+    elemManager->ConstructViewAccessor<array1d<real64>, arrayView1d<real64>>( viewKeyStruct::deltaPressureString );
+
+  m_temperature =
+    elemManager->ConstructViewAccessor<array1d<real64>, arrayView1d<real64>>( viewKeyStruct::temperatureString );
+  m_deltaTemperature =
+    elemManager->ConstructViewAccessor<array1d<real64>, arrayView1d<real64>>( viewKeyStruct::deltaTemperatureString );  
+
+  m_concentration =
+    elemManager->ConstructViewAccessor<array2d<real64>, arrayView2d<real64>>( viewKeyStruct::concentrationString );
+  m_deltaConcentration =
+    elemManager->ConstructViewAccessor<array2d<real64>, arrayView2d<real64>>( viewKeyStruct::deltaConcentrationString );
+  m_totalConcentration =
+    elemManager->ConstructViewAccessor<array2d<real64>, arrayView2d<real64>>( viewKeyStruct::totalConcentrationString );  
+
+  m_concentrationNew =
+    elemManager->ConstructViewAccessor<array2d<real64>, arrayView2d<real64>>( viewKeyStruct::concentrationNewString );
+
+  
+//  m_dependentConc =
+//    elemManager->ConstructFullMaterialViewAccessor<array2d<real64>, arrayView2d<real64> >( ReactiveFluidBase::viewKeyStruct::dependentConcString, constitutiveManager );
+//
+//  m_dDependentConc_dConc =
+//    elemManager->ConstructFullMaterialViewAccessor<array3d<real64>, arrayView3d<real64> >( ReactiveFluidBase::viewKeyStruct::dDependentConc_dConcString, constitutiveManager );
```

Comment:
@klevzoff You reworked this at some point, and I am not sure of the correct usage. Can you suggest the appropriate code to replace this?

Reply:
This is github-ware, I hope it compiles
```suggestion
  m_dependentConc =
    elemManager->ConstructMaterialArrayViewAccessor< real64, 2 >( ReactiveFluidBase::viewKeyStruct::dependentConcString, 
                                                                  targetRegionNames(),
                                                                  fluidModelNames() );

  m_dDependentConc_dConc =
    elemManager->ConstructMaterialArrayViewAccessor< real64, 3 >( ReactiveFluidBase::viewKeyStruct::dDependentConc_dConcString, 
                                                                  targetRegionNames(),
                                                                  fluidModelNames() );
```